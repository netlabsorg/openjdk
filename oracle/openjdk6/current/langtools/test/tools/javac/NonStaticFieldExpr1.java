/* @test  /nodynamiccopyright/
   @bug 4087127 4785453
   @author dps
   @summary field: instance access through types is not allowed

   @run shell NonStaticFieldExpr1.sh
*/
class NonStaticFieldExpr1 {
  public int x;
  int y = NonStaticFieldExpr1.x;                // SHOULD BE ERROR
}
