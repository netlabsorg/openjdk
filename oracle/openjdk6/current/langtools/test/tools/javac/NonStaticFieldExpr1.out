NonStaticFieldExpr1.java:10: non-static variable x cannot be referenced from a static context
  int y = NonStaticFieldExpr1.x;                // SHOULD BE ERROR
                             ^
1 error
